lib/App/Config.pm

```diff
@@ -0,0 +1,422 @@
+    $self->_add_dynamic_setting_info($attribute->path, $definition);
+
+    return;
+}
+
+sub _create_generic_attribute {
+    my $self       = shift;
+    my $section    = shift;
+    my $name       = shift;
+    my $definition = shift;
+
+    $self->_add_attribute('App::Config::Attribute', $section, $name, $definition);
+
+    return;
+}
+
+sub _add_attribute {
+    my $self       = shift;
+    my $attr_class = shift;
+    my $section    = shift;
+    my $name       = shift;
+    my $definition = shift;
+
+    my $fake_name = "a_$name";
+    my $writer    = "_$fake_name";
+
+    my $attribute = $attr_class->new(
+        name        => $name,
+        definition  => $definition,
+        parent_path => $section->path,
+        data_set    => $self->data_set,
+    )->build;
+
+    $section->meta->add_attribute(
+        $fake_name,
+        is      => 'ro',
+        handles => {
+            $name          => 'value',
+            'has_' . $name => 'has_value',
+        },
+        documentation => $definition->{description},
+        writer        => $writer,
+    );
+
+    $section->$writer($attribute);
+
+    return $attribute;
+}
+
+sub _validate_key {
+    my $self    = shift;
+    my $key     = shift;
+    my $section = shift;
+
+    if (grep { $key eq $_ } qw(path parent_path name definition version data_set check_for_update save_dynamic)) {
+        die "Variable with name $key found under "
+            . $section->path
+            . ".\n$key is an internally used variable and cannot be reused, please use a different name";
+    }
+
+    return;
+}
+
+=head2 check_for_update
+
+check and load updated settings from chronicle db
+
+=cut
+
+sub check_for_update {
+    my $self     = shift;
+    my $data_set = $self->data_set;
+
+    my $app_settings = $self->chronicle_reader->get($self->setting_namespace, $self->setting_name);
+
+    if ($app_settings and $data_set) {
+        my $db_version = $app_settings->{_rev};
+        unless ($data_set->{version} and $db_version and $db_version eq $data_set->{version}) {
+            $self->_add_app_setttings($data_set, $app_settings);
+        }
+    }
+
+    return;
+}
+
+=head2 save_dynamic
+
+Save synamic settings into chronicle db
+
+=cut
+
+sub save_dynamic {
+    my $self = shift;
+    my $settings = $self->chronicle_reader->get($self->setting_namespace, $self->setting_name) || {};
+
+    #Cleanup globals
+    my $global = Data::Hash::DotNotation->new();
+    foreach my $key (keys %{$self->dynamic_settings_info->{global}}) {
+        if ($self->data_set->{global}->key_exists($key)) {
+            $global->set($key, $self->data_set->{global}->get($key));
+        }
+    }
+
+    $settings->{global} = $global->data;
+    $settings->{_rev}   = time;
+    $self->chronicle_writer->set($self->setting_namespace, $self->setting_name, $settings);
+
+    return 1;
+}
+
+=head2 current_revision
+
+get current revision of settings.
```

Comment:
Please, mention that it actually loads settings from chronicles, returns the last revision and  drops them